examples/binaries/init-with-value/src/lib.rs

```diff
@@ -60,20 +90,41 @@ mod wasm {
         let data: gstd::Vec<SendMessage> = msg::load().expect("provided invalid payload");
         for msg_data in data {
             match msg_data {
-                SendMessage::Init(value) => {
+                SendMessage::Init {
+                    value,
+                    expected_error,
+                } => {
                     let submitted_code = CHILD_CODE_HASH.into();
-                    let _ = prog::create_program_with_gas(
+                    let res = prog::create_program_with_gas(
                         submitted_code,
                         COUNTER.to_le_bytes(),
                         [],
                         1_000_001,
                         value,
                     );
 
-                    COUNTER += 1;
+                    match res {
+                        Ok(_) => {
+                            COUNTER += 1;
+                        }
+                        Err(err) => {
+                            gstd::debug!("Error occurred during program creation: {}", err);
+                            if let Some(expected_error) = expected_error {
+                                assert_eq!(
+                                    err,
+                                    ContractError::Ext(ExtError::Message(
+                                        expected_error.into_message_error()
+                                    ))
+                                )
+                            }
+                        }
+                    }
```

Comment:
Although idea is quite original, but I suggest the next thing. that seems a bit easier: 
1. on error send a reply to user with an actual `ContractError::Ext(...)` (or could be just `MessageError`, or it's string format) error encoded as a payload.
2. check in the pallet test whether the message with expected error payload exists.

One more reason for such changes is to avoid dirty conversions from `ExpectedError` to `MessageError` using hard-coded values, which can cause problems in the future

Reply:
What do you think about current realisation? Error is created on pallet side and checked on contract's one